refactor(web-diary): css values set using theme

gochicus opened the pull request:
affects: @web-diary/app, @components/form-post-add, @components/header, @practice/web-diary,
@ui/list, @ui/theme

ISSUES CLOSED: #72

gochicus committed:
refactor(web-diary): css values set using theme

affects: @web-diary/app, @components/form-post-add, @components/header, @practice/web-diary,
@ui/list, @ui/theme

ISSUES CLOSED: #72

TFK70 commented on practice/web-diary/components/form-post-add/src/FormPostAdd.tsx:
```diff
@@ -5,6 +5,7 @@ import { Button }           from '@ui/button'
 import { Input }            from '@ui/input'
 import { Box, Layout, Row } from '@ui/layout'
 import { usePostData }      from '@store/post-data'
+import { theme }            from '@ui/theme'
```
[Комментарии 1 в 1](https://github.com/atls-academy/frontend-study-ikorol/pull/79)

gochicus committed:
Merge branch 'master' into refactor/use-theme

gochicus committed:
refactor(web-diary): use of theme refactored

affects: @web-diary/app, @components/form-post-add, @components/header, @components/post-list,
@components/switcher-post-status, @practice/web-diary, @ui/list, @ui/theme

ISSUES CLOSED: #72

TFK70 commented on practice/web-diary/ui/theme/src/minWidth.ts:
```diff
@@ -0,0 +1,6 @@
+export const minWidth = {
```
такие вещи в тему нет большого смысла выносить
значений для minWidth у тебя может быть миллиард самых разных, и это нормально
другое дело постоянно `#F9A0D8` писать вместо `pink`

TFK70 commented on practice/web-diary/app/src/App.tsx:
```diff
@@ -1,16 +1,16 @@
+import { SearchValueProvider }     from '@store/search-status'
+import { Background }              from '@ui/background'
+import { Column, Layout }          from '@ui/layout'
+import { color, indent, minWidth } from '@ui/theme'
```
\* as theme

TFK70 commented on practice/web-diary/app/src/App.tsx:
```diff
@@ -28,12 +33,12 @@ const App = () => {
-                <Layout flexBasis={theme.indent.normal} />
-                <FormPostAdd />
+                <Layout flexBasis={indent.normal} />
+                <SwitcherPostStatus theme={theme} />
```
а провайдер тогда зачем?

TFK70 commented on practice/web-diary/components/switcher-post-status/src/SwitcherPostStatus.tsx:
```diff
@@ -9,31 +9,31 @@ import { useSearchValue }   from '@store/search-status'
-        <Layout flexBasis='15px' />
-        <Button color='black' onClick={() => setStatus('all')}>
+        <Layout flexBasis={theme.indent.small} />
+        <Button color={theme.color.black} onClick={() => setStatus('all')}>
```
тот комментарий про фичу styled-system не только к fontSize относится

TFK70 commented on practice/web-diary/components/switcher-post-status/src/SwitcherPostStatus.tsx:
```diff
@@ -9,31 +9,31 @@ import { useSearchValue }   from '@store/search-status'
         />
-        <Layout flexBasis='15px' />
-        <Button color='black' onClick={() => setStatus('all')}>
+        <Layout flexBasis={theme.indent.small} />
```
ну и вот тут должно быть число, такое не надо выносить в тему, как я уже сказал

TFK70 commented on practice/web-diary/components/switcher-post-status/src/SwitcherPostStatus.tsx:
```diff
@@ -9,31 +9,31 @@ import { useSearchValue }   from '@store/search-status'
 import messages             from '../messages/messages'
 
-export const SwitcherPostStatus = () => {
+export const SwitcherPostStatus = ({ theme }) => {
```
а после правок тебе тут вообще объект с темой не понадобится т.к. компоненты будут брать значения оттуда внутри себя

gochicus committed:
refactor(web-diary): useof theme refactored

affects: @web-diary/app, @components/form-post-add, @components/header, @components/post-list,
@components/switcher-post-status, @practice/web-diary, @ui/list, @ui/theme

ISSUES CLOSED: #72

TFK70 commented on practice/web-diary/components/header/src/Header.tsx:
```diff
@@ -9,15 +10,16 @@ import messages            from '../messages/messages'
   return (
     <Row justifyContent='space-between' alignItems='flex-end'>
-      <Text fontSize='basic'>{postData.length}</Text>
+      <Text fontSize={theme.fontSize.basic}>{postData.length}</Text>
```
Объект с размерами шрифтов должен называться fontSizes
Переименуй и все заработает

Ну и так по аналогии со всем остальным: colors, fontWeights и тд

gochicus committed:
refactor(web-diary): object renamed

affects: @components/header, @practice/web-diary, @ui/theme

ISSUES CLOSED: #72
